### PR TITLE
Add GCP.IAM.serviceAccounts.getAccessToken.Privilege.Escalation rule

### PR DESCRIPTION
**Pull request by melenevskyi**
### Background
The iam.serviceAccounts.signBlob permission “allows signing of arbitrary payloads” in GCP. This means we can create a signed blob that requests an access token from the Service Account we are targeting.
Should be merged after or instead of https://github.com/panther-labs/panther-analysis/pull/1148

### Changes

Add GCP.IAM.serviceAccounts.getAccessToken.Privilege.Escalation rule

### CODE REVIEW

File: rules/gcp_audit_rules/gcp_iam_service_accounts_sign_blob.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from gcp_base_helpers import gcp_alert_context
+from panther_base_helpers import deep_walk
```
Comment (arielkr256) [inline rules/gcp_audit_rules/gcp_iam_service_accounts_sign_blob.py:2]: `event.deep_walk("field"...`

**Review by arielkr256 — CHANGES_REQUESTED** [review]

File: rules/gcp_audit_rules/gcp_iam_service_accounts_sign_blob.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    authorization_info = deep_walk(event, "protoPayload", "authorizationInfo")
+    if not authorization_info:
+        return False
+
+    for auth in authorization_info:
+        if auth.get("permission") == "iam.serviceAccounts.signBlob" and auth.get("granted") is True:
+            return True
+    return False
+
+
```
[the remark this answers is not in the document]
Comment (melenevskyi) [reply rules/gcp_audit_rules/gcp_iam_service_accounts_sign_blob.py:15]: why? there is `DisplayName` in yaml and it used in generated alert https://threat-research.runpanther.net/alerts-and-errors/67d87eb3b984610f34cd8008b5d33e3f/

File: rules/gcp_audit_rules/gcp_iam_service_accounts_sign_blob.py (hunk 1 above)
Comment (melenevskyi) [reply rules/gcp_audit_rules/gcp_iam_service_accounts_sign_blob.py:2]: could you elaborate?)

Comment (melenevskyi) [reply rules/gcp_audit_rules/gcp_iam_service_accounts_sign_blob.py:2]: I guess you meant this https://github.com/panther-labs/panther-analysis/pull/1148#discussion_r1525857290

File: rules/gcp_audit_rules/gcp_iam_service_accounts_sign_blob.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (arielkr256) [reply rules/gcp_audit_rules/gcp_iam_service_accounts_sign_blob.py:15]: DisplayName is a static description of what the rule does, the title function is dynamic for each alert triggered and should include more specific information such as actor name, etc.

**Review by arielkr256 — APPROVED** [review]

**Pull request merged**